README.md

```diff
@@ -1,81 +1,148 @@
-Offering this in bpfd allows for XDP applications whose loader was not using libxdp to benefit from this.
-We are also hoping to find a way for applications linked with libxdp to use bpfd instead if it's
-in use in the system.
-
-## Usage
-
-Load a sample XDP Program:
-
-```console
-cargo build
-sudo ./target/debug/bpfd&
-sudo ./target/debug/bpfctl load --path /path/to/xdp/program --section-name "pass" xdp -i wlp2s0 --priority 50
-```
+bpfd is a system daemon for managing BPF programs that works both on a single
+node and in a Kubernetes cluster using the bpfd operator and CRDs.
+
+## Why BPF?
+
+BPF is a powerful general-purpose framework that allows running sandboxed
+programs in the kernel.  It can be used for many purposes, including networking,
+monitoring, tracing and security.
+
+## Why BPF in Kubernetes?
+
+Demand is increasing from both Kubernetes developers and users. Examples of BPF
+in Kubernetes include:
+
+- [Cilium](https://cilium.io/) and
+  [Calico](https://www.tigera.io/project-calico/) CNIs 
+- [Pixie](https://px.dev/): Open source observability
+- [KubeArmor](https://kubearmor.io/): Container-aware runtime security
+  enforcement system
+- [Blixt](https://github.com/Kong/blixt): Gateway API L4 conformance
+  implementation
+- [NetObserv](https://github.com/netobserv): Open source operator for network
+  observability 
+
+## Challenges for BPF in Kubernetes
+
+- Requires privileged pods
+  - BPF-enabled apps require at least CAP_BPF permissions and potentially more
+    depending on the type of program that is being attached.
+  - Since the Linux capabilities are very broad it is challenging to constrain a
+    pod to the minimum set of privileges required. This can allow them to do
+    damage (either unintentionally or intentionally).
+- Handling multiple BPF programs on the same BPF hooks
+  - Not all BPF hooks are designed to support multiple programs.
+  - Some software using BPF assumes exclusive use of a BPF hook and can
+    unintentionally eject existing programs when being attached.
+- Debugging problems with deployments is hard.
+  - The cluster administrator may not be aware that BPF programs are being used
+    in a cluster.
+  - It is possible for some BPF programs to interfere with others in
+    unpredictable ways.
+  - SSH access or a privileged pod is necessary to determine the state of BPF
+    programs on each node in the cluster.
+- There are no standard libraries for loading and managing BPF programs leaving
+  each application developer to implement their own.
+- Deployment on Kubernetes is not simple.
+	  - It is an involved process that requires first writing a daemon that loads
+	    your BPF bytecode and deploying it using a DaemonSet.
+	  - This requires careful design and intricate knowledge of the BPF program
+	    lifecycle to ensure your program stays loaded and that you can easily
+	    tolerate pod restarts and upgrades.
+
```

Comment:
What about fine grain versioning problems? 

Something along the lines of....

- In Bpf enabled K8s deployments today the Bpf Program is often embedded into the userspace binary that loads and interacts with it.  This means there's no easy way to have fine-grained versioning control  of the bpfProgram in relation to it's accompanying userspace counterpart 

Or something like that :)